### PR TITLE
[ISSUE-127] Only change password and password_hash when value changes

### PR DESCRIPTION
**Pull request by RobsonSutton**
- Only update `password` and `password_hash` when the value has changed in the Terraform configuration (identified by: https://github.com/elastic/terraform-provider-elasticstack/issues/127) (required behaviour in issue is backed up by API docs: https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-put-user.html)

### CODE REVIEW

**Comment by tobio** [comment]
jenkins test this

**Comment by tobio** [comment]
jenkins test this

**Comment by tobio** [comment]
jenkins test this

**Comment by tobio** [comment]
@dimuon are you able to take a look at this one too?

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 - Correctly identify a missing security user ([#101](https://github.com/elastic/terraform-provider-elasticstack/issues/101))
 - Support **7.x** Elasticsearch < **7.15** by removing the default `media_type` attribute in the Append processor ([#118](https://github.com/elastic/terraform-provider-elasticstack/pull/118))
 - Add `allow_restricted_indices` setting to security role ([#125](https://github.com/elastic/terraform-provider-elasticstack/issues/125))
+- Add conditional to only set `password` and `password_hash` when a new value is defined ([#127](https://github.com/elastic/terraform-provider-elasticstack/pull/128))
```
Comment (dimuon) [inline CHANGELOG.md:10]: Does it make sense to mention getting default value for `insecure` attribute from `ELASTICSEARCH_INSECURE` env var?

File: internal/elasticsearch/security/user.go [hunk 2]
```diff
@@ -115,13 +115,18 @@ func resourceSecurityUserPut(ctx context.Context, d *schema.ResourceData, meta i
 
 	var user models.User
 	user.Username = usernameId
+
 	if v, ok := d.GetOk("password"); ok {
 		password := v.(string)
-		user.Password = &password
+		if d.HasChange("password") {
```
Comment (dimuon) [inline internal/elasticsearch/security/user.go]: Does it make sense to simplify logic a bit, e.g. remove the embedded `if` and move `d.HasChange("password")` to line 119?
Same with the `password_hash`.

**Review by dimuon — DISMISSED** [review]
LGTM

**Comment by tobio** [comment]
jenkins test this

**Review by dimuon — APPROVED** [review]

**Pull request merged**